site/docs/relations/logical_relations.md

```diff
@@ -229,8 +229,11 @@ The join operation will combine two separate inputs into a single output, based
 | Left  | Return all records from the left input. For each cross input match, return a record including the data from both sides. For any remaining non-matching records from the left input, return the left record along with nulls for the right input. |
 | Right | Return all records from the right input. For each cross input match, return a record including the data from both sides. For any remaining non-matching records from the right input, return the right record along with nulls for the left input. |
 | Semi | Returns records from the left input. These are returned only if the records have a join partner on the right side. |
+| Right Semi | Returns records from the right input. These are returned only if the records have a join partner on the left side. |
 | Anti  | Return records from the left input. These are returned only if the records do not have a join partner on the right side. |
-| Single | Returns one join partner per entry on the left input. If more than one join partner exists, there are two valid semantics. 1) Only the first match is returned. 2) The system throws an error. If there is no match between the left and right inputs, NULL is returned. |
+| Right Anti  | Return records from the right input. These are returned only if the records do not have a join partner on the left side. |
+| Left Single | Return all records from the left input with no join expansion. If at least one record from the right input matches the primary, return one arbitrary matching record from the right input. For any left records without matching right records, return the primary (left) record along with nulls for the right input. Similar to a left outer join but only returns one right match at most. Useful for nested sub-queries where we need exactly one record in output (or throw exception).  See Section 3.2 of https://15721.courses.cs.cmu.edu/spring2018/papers/16-optimizer2/hyperjoins-btw2017.pdf for more information. |
```

Comment:
missed changing some of the primary references.

Reply:
Removed those.  Didn't touch the original use of primary in the set portion of the documentation.